### PR TITLE
Fix the references for VP8 encoding

### PR DESCRIPTION
**Pull request by xhaihao**
If a reference is unavailable, use the next frame instead

### CODE REVIEW

File: src/i965_encoder_vp8.c [hunk 1]
```diff
@@ -2087,6 +2087,19 @@ i965_encoder_vp8_get_picture_parameter(VADriverContextP ctx,
         vp8_context->ref_arf_frame = NULL;
     }
 
+    if (!is_intra) {
+        if (!vp8_context->ref_last_frame) {
+            if (vp8_context->ref_gf_frame)
+                vp8_context->ref_last_frame = vp8_context->ref_gf_frame;
+            else
+                vp8_context->ref_last_frame = vp8_context->ref_arf_frame;
```
Comment (lizhong1008) [inline src/i965_encoder_vp8.c]: Is it possible last/gf/arf frames are all NULL frame? Maybe we can add an assert() for this case.

File: src/i965_encoder_vp8.c [hunk 2]
```diff
@@ -2087,6 +2087,19 @@ i965_encoder_vp8_get_picture_parameter(VADriverContextP ctx,
...
+        }
+
+        if (!vp8_context->ref_gf_frame) {
+            vp8_context->ref_gf_frame = vp8_context->ref_arf_frame;
```
Comment (lizhong1008) [inline src/i965_encoder_vp8.c]: Should check ref_arf_frame is not NULL too?

File: src/i965_encoder_vp8.c [hunk 3]
```diff
@@ -5946,14 +5959,26 @@ i965_encoder_vp8_pak_pipeline_prepare(VADriverContextP ctx,
...
+    }
+
+    if (vp8_context->ref_arf_frame) {
+        PAK_REFERENCE_BO(vp8_context->reference_surfaces[2].bo, vp8_context->ref_arf_frame->bo, 1);
+    } else {
+        PAK_REFERENCE_BO(vp8_context->reference_surfaces[2].bo, NULL, 0);
+    }
+
+    for (i = 3; i < ARRAY_ELEMS(vp8_context->reference_surfaces); i++) {
+        PAK_REFERENCE_BO(vp8_context->reference_surfaces[i].bo, NULL, 0);
```
Comment (lizhong1008) [inline src/i965_encoder_vp8.c]: The reset LGTM

File: src/i965_encoder_vp8.c (hunk 1 above)
Comment (xhaihao) [reply src/i965_encoder_vp8.c]: For non-key frame, one of last/gf/arf must be non-NULL.

Comment (lizhong1008) [reply src/i965_encoder_vp8.c]: This is normal case. Actually for normal case, last frame must be non-NULL for P frames.

**Comment by uartie** [comment]
What is the status here?

**Comment by xhaihao** [comment]
@uartie Some extra works are required, currently it is in low priority.

**Pull request closed without merging**